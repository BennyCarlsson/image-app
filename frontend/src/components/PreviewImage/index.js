import React from 'react'
import { Img } from './styles'

const PreviewImage = ({ src }) => {
  return <Img alt='Preview' src={src} />
}

export default PreviewImage
