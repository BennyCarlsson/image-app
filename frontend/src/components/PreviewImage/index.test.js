import React from 'react'
import { render, screen } from '@testing-library/react'
import PreviewImage from '.'

describe('<PreviewImage />', () => {
  const src = 'test.jpg'

  it('should match snapshot', () => {
    const { asFragment } = render(<PreviewImage src={src} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    render(<PreviewImage src={src} />)
    expect(screen).toBeTruthy()
  })

  it('should render img with src', () => {
    render(<PreviewImage src={src} />)
    expect(screen.getByAltText('Preview')).toHaveAttribute('src', src)
  })
})
