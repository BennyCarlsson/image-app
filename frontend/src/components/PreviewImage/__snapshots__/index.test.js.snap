// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewImage /> should match snapshot 1`] = `
<DocumentFragment>
  <img
    alt="Preview"
    class="sc-bdvvtL kJkrPP"
    src="test.jpg"
  />
</DocumentFragment>
`;
